Mod2/Lessons/Week3/SQLJoins.md

```diff
@@ -0,0 +1,280 @@
+
+id |     name      | hometown     | years_active | id |      title      | length     | play_count | artist_id
+---+---------------+--------------+--------------+----+-----------------+------------+------------+----------
+ 1 | Prince        |  Minneapolis | 1975-2016    | 1  | Purple Rain     |        345 |        23  |         1
+ 1 | Prince        |  Minneapolis | 1975-2016    | 2  | Raspberry Beret |        432 |        12  |         1
+ 2 | Talking Heads |  New York    | 1975-1991    | 3  | Wild Wild Life  |        367 |       367  |         2
+(3 rows)
+```
+
+**Turn and Talk** What did this query do? How might you describe the return value of this query?
+
+When we run this `JOIN`, we are _joining_ the songs and artists tables together to return a table that includes all the information from _both_ tables. For each artist, we see a row for each song that they have, with the information from both the artists table and the songs table.
+
+When creating a `JOIN` query, there are three essential parts:
+
+  1. `SELECT` - this is what indicates which columns will be included in the resulting table
+      - `SELECT * FROM songs`
+  2. `JOIN` - the command to join in data from another table
+      - `JOIN artists`
+  3. `ON` - this tells the join _how_ to join the two tables together, or what is the relationship between the two tables (most often, primary key = foreign key)
+      - `ON artists.id = songs.artist_id`
+
+Let's now add another artist, Zac Brown Band.
+
+```SQL
+INSERT INTO 
+	artists ( 
+		name, hometown, years_active
+	)
+VALUES
+	('Zach Brown Band','Atlanta','2002-Present');
+```
+
+> **With Your Partner** Run the `JOIN` again, `Zac Brown Band` is not returned in the data! Why are we not seeing _all_ our artists on this joined table?
+
+
+
+Let's now add a song where we don't know the artist info.
+
+```SQL
+INSERT INTO 
+	songs ( 
+		title, length, play_count
+	)
+VALUES
+	('Wow such a great song', 10, 1000);
+```
+Do you think this data will be returned when we run our `JOIN`? Let's pause and predict and then run it!
+
+<!-- Add dropdown spoiler -->
+
+<details><summary>Spoiler</summary><br/>
+<p align='center'>
+The song without an artist was <b>not</b> returned in the join.
+</p>
+</details>
+
+
+# Venn Diagram Review
+When we create `JOIN` queries, there are a handful of different join types that we can declare that will affect the resulting table. We can use Venn Diagrams to represent what data will be returned for each type of join.
+
+Let's review the images you looked at in preparation for this lesson.
+
+<p align='center'>
+  <img src='../../Images/Week3/venn_diagrams.png'>
+</p>
+
+> **With Your Breakout Room** Explain what each of the shaded regions represents. What animals did you put in each region?
+
+<!-- Instructor note, it would be awesome for students to annotate and add animals to the image or use pair deck or something here. -->
```

Comment:
I agree, this can be a fun activity. We might want to curate a cache of animal pictures for students to drag on a Jamboard or something (just throwing out an idea).

Reply:
Love that idea! Not going to venture into Jamboard now, but will add as a note for whoever ends up leading this lesson to consider.